crates/fuel-core/src/state/rocks_db.rs

```diff
@@ -313,18 +314,30 @@ where
 
         let existing_column_families = DB::list_cf(&opts, &path).unwrap_or_default();
 
-        let mut cf_descriptors_to_open = vec![];
-        let mut cf_descriptors_to_create = vec![];
+        let mut cf_descriptors_to_open = BTreeMap::new();
+        let mut cf_descriptors_to_create = BTreeMap::new();
         for column in columns.clone() {
             let column_name = Self::col_name(column.id());
             let opts = Self::cf_opts(column, &block_opts);
             if existing_column_families.contains(&column_name) {
-                cf_descriptors_to_open.push((column_name, opts));
+                cf_descriptors_to_open.insert(column_name, opts);
             } else {
-                cf_descriptors_to_create.push((column_name, opts));
+                cf_descriptors_to_create.insert(column_name, opts);
             }
         }
 
+        for column_name in existing_column_families {
+            if cf_descriptors_to_open.contains_key(&column_name)
+                || cf_descriptors_to_create.contains_key(&column_name)
+            {
+                continue;
+            }
+
+            let unknown_column_name = column_name;
+            let unknown_column_options = Self::default_cf_opts(&block_opts);
+            cf_descriptors_to_open.insert(unknown_column_name, unknown_column_options);
+        }
```

Comment:
While this PR is merged without applying this suggestion, I also generally prefer using iterators to populate collections over using for loops and `.insert`. In this scenario I would also have advised to use `.map` instead of `.for_each`, to avoid having to call `.insert` from within the for_each closure.

Something along the lines of:
```
cf_descriptors_to_open.extend(existing_column_families.iter().filter(...).map(...))
```